circle_evolution/main.py

```diff
@@ -9,20 +9,23 @@
 import circle_evolution.helpers as helpers
 
 
+SIZE_OPTIONS = {1: (64, 64), 2: (128, 128), 3: (256, 256), 'auto': None}
+
+
 def main():
     """Entrypoint of application"""
     parser = argparse.ArgumentParser(description="Circle Evolution CLI")
 
     parser.add_argument("image", type=str, help="Image to be processed")
-    size_options = {1: (64, 64), 2: (128, 128), 3: (256, 256)}
-    parser.add_argument("--size", choices=size_options.keys(), default=2, help="Dimension of the image", type=int)
+    parser.add_argument("--size", choices=SIZE_OPTIONS.keys(), default='auto', help="Dimension of the image")
     parser.add_argument("--genes", default=256, type=int, help="Number of genes")
     parser.add_argument("--max-generations", type=int, default=500000)
     args = parser.parse_args()
 
     target = helpers.load_target_image(args.image, size=size_options[args.size])
 
-    evolution = Evolution(size_options[args.size], target, genes=args.genes)
+    output_img_dimensions = size_options[args.size] or target.shape
```

Comment:
There is an error here, you forgot to put the right name `SIZE_OPTIONS`.

Reply:
Wooops, fixed